src/flash.js

```diff
@@ -621,6 +660,162 @@ class Flash extends FakeEventTarget implements IEngine {
+
+  /**
+   * Gets an image to be shown while the video is downloading, or until the user hits the play button.
+   * @returns {poster} - The image url.
+   * @public
+   */
+  get poster(): string {
+    return '';
+  }
+
+  /**
+   * Specifies if and how the author thinks that the video should be loaded when the page loads.
+   * @param {string} preload - The preload value.
+   * @public
+   * @returns {void}
+   */
+  set preload(preload: string): void {}
+
+  /**
+   * Gets the preload value of the video element.
+   * @returns {string} - The preload value.
+   * @public
+   */
+  get preload(): string {
+    return 'none';
+  }
+
+  /**
+   * Set if the video will automatically start playing as soon as it can do so without stopping.
+   * @param {boolean} autoplay - The autoplay value.
+   * @public
+   * @returns {void}
+   */
+  set autoplay(autoplay: boolean): void {}
+
+  /**
+   * Gets the autoplay value of the video element.
+   * @returns {boolean} - The autoplay value.
+   * @public
+   */
+  get autoplay(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that video controls should be displayed.
+   * @param {boolean} controls - the controls value.
+   * @public
+   * @returns {void}
+   */
+  set controls(controls: boolean): void {}
+
+  /**
+   * Gets the controls value of the video element.
+   * @returns {boolean} - The controls value.
+   * @public
+   */
+  get controls(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that the video will start over again, every time it is finished.
+   * @param {boolean} loop - the loop value.
+   * @public
+   * @returns {void}
+   */
+  set loop(loop: boolean) {}
+
+  /**
+   * Gets the loop value of the video element.
+   * @returns {boolean} - The loop value.
+   * @public
+   */
+  get loop(): boolean {
+    return false;
+  }
+
+  get isInPictureInPicture(): boolean {
+    return false;
+  }
+
+  /**
+   * @returns {Number} - The current network state (activity) of the audio/video.
+   * @public
+   */
+  get networkState(): number {
+    return 1;
+  }
+
+  /**
+   * Indicates if the audio/video is ready to play or not.
+   * @returns {Number} - The current ready state of the audio/video.
+   * 0 = HAVE_NOTHING - no information whether or not the audio/video is ready.
+   * 1 = HAVE_METADATA - metadata for the audio/video is ready.
+   * 2 = HAVE_CURRENT_DATA - data for the current playback position is available, but not enough data to play next frame/millisecond.
+   * 3 = HAVE_FUTURE_DATA - data for the current and at least the next frame is available.
+   * 4 = HAVE_ENOUGH_DATA - enough data available to start playing.
+   */
+  get readyState(): number {
+    if (!this._api) {
+      return 0;
+    }
+    return 4;
+  }
+
+  /**
+   * @returns {Number} - The height of the video player, in pixels. flash object will take 100 of this element
+   * @public
+   */
+  get videoHeight(): number {
+    return this._el.getBoundingClientRect().height;
```

Comment:
Is there no flash HLS api to get this info?

Reply:
No, it takes 100% from this element

Reply:
Please read what it means in HTML5 api https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement/videoWidth
I meant to check if we can get this real Intrinsic width and height and not the doc width and height.
If we can’t get it then return like youtube.